### DISCUSSION TITLE
What is the correct way to upgrade pdk monorepo dependencies?

### DISCUSSION THREAD

**Discussion by janeklb** [aws/aws-pdk#644] [Q&A] [answered]
I have a pdk (typescript) monorepo project with a handful sub-projects (in the `packages/` directory). I'm using `pnpm`.

What's the right way to upgrade all the dependencies in the monorepo? Afaict there are a few options:

- `pdk run upgrade-deps` (will run the `upgrade-deps` projen task, via `aws-pdk`'s monorepo project)
- `pdk run upgrade` (will run the `upgrade` projen task)
- `pnpm update --recursive` (will run the `pnpm`-native update process)

It seems like `upgrade-deps` is the right way to do it, but it's a little confusing to see an `upgrade` projen task also available. Interestingly `upgrade-deps` *does not* upgrade `pdk` itself..

Furthermore, confusion is exacerbated by the fact that if you run `pdk upgrade` (or `update` or `up`), as opposed to `pdk run upgrade`, you'll end up triggering `pnpm`'s `upgrade`.
I think there was a recent change that allowed you to `cdk deploy` via `pdk deploy` (instead of `pdk run deploy`) when using `pnpm` since `pnpm` has it's own `deploy` command.. but this is probably out of scope for pdk since the `upgrade` task comes from `projen`.

**Comment by agdimech** [comment, marked as answer]
Hi - `upgrade-deps` is the way to go from a monorepo perspctive as it will update all packages within the monorepo.

`upgrade` on the other hand will only update packages for the current package it is executed from. Probably a poor choice of naming from my perspective so I apologize for that. We should probably also add supporting documentation as this is lacking at the moment.

In relation to pdk itself not upgrading, do you mean your dependencies on `@aws/pdk`?

The pnpm upgrade issue is definitely a bug that needs to be addressed. Let me go ahead and create an issue to track these here: https://github.com/aws/aws-pdk/issues/653

I will close this discussion and ask that you please respond to my questions in the linked issue :)
